feature: RP intermediary close to EUDIW from fork (#1140)
italia/eidas-it-wallet-docs: docs/en/trust-infrastructure.rst

     - |check-icon|

Trust Marks for Relying Party Intermediaries
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> In the IT-Wallet ecosystem, Intermediaries acting on behalf of Relying Parties (pursuant to Art. 5b(8) of the eIDAS2 Regulation `EU_2024_1183`_) are subject to a specific Trust Mark regime that allows the Wallet to recognize and display to the User that the requesting RP operates through an Intermediary. In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_).

Reviewer: ```suggestion
In the IT-Wallet ecosystem, Relying Party Intermediaries acting on behalf of Relying Parties (pursuant to Art. 5b(8) of the eIDAS2 Regulation `EU_2024_1183`_) are subject to a specific Trust Mark regime that allows the Wallet to recognize and display to the User that the requesting RP operates through an Intermediary. In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_).
```

> In the IT-Wallet ecosystem, Intermediaries acting on behalf of Relying Parties (pursuant to Art. 5b(8) of the eIDAS2 Regulation `EU_2024_1183`_) are subject to a specific Trust Mark regime that allows the Wallet to recognize and display to the User that the requesting RP operates through an Intermediary. In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_).

Reviewer: This sentence is repeated in several places.
> In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_).
>

Author: Done. :).

> In the IT-Wallet ecosystem, Intermediaries acting on behalf of Relying Parties (pursuant to Art. 5b(8) of the eIDAS2 Regulation `EU_2024_1183`_) are subject to a specific Trust Mark regime that allows the Wallet to recognize and display to the User that the requesting RP operates through an Intermediary. In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_): it participates in the federation trust chain as an Intermediate, publishing its own Entity Configuration and issuing Subordinate Statements for its affiliated Relying Parties (see the definition of **IT-Wallet Intermediary**).

Reviewer 2: ```suggestion
In the IT-Wallet ecosystem, Intermediaries acting on behalf of Relying Parties (pursuant to Art. 5b(8) of the eIDAS2 Regulation `EU_2024_1183`_) are subject to a specific Trust Mark regime that allows the Wallet to recognize and display to the User that the requesting RP operates through an Intermediary. In the implementation profile described by these technical specifications, a Relying Party Intermediary is also an OpenID Federation Intermediate Entity (`OID-FED`_).
```

**Trust Mark Issuance to the Intermediary**

> The Trust Anchor MUST include Relying Party Intermediaries in the ``trust_mark_issuers`` attribute of its own Entity Configuration, authorizing them to issue Trust Marks for their subordinate Leaves. The Trust Anchor issues a Trust Mark to the Intermediary attesting its role, using ``https://<federation_authority_domain>/trust_marks/federation-entity/rp_intermediary`` as ``trust_mark_type`` to distinguish it from the other entities in the ecosystem.

Reviewer 2: è ripetuto, parte di questo è stato già detto sopra

```suggestion
The Trust Anchor MUST include Relying Party Intermediaries in the ``trust_mark_issuers`` attribute of its own Entity Configuration, authorizing them to issue Trust Marks for their subordinate Leaves. The Trust Anchor issues a Trust Mark to the Intermediary attesting its role, using ``https://<federation_authority_domain>/trust_marks/federation-entity/openid_credential_verifier_intermediary`` as ``trust_mark_type`` to distinguish it from the other entities in the ecosystem.
```

The Relying Party Intermediary MUST include this Trust Mark in its own Entity Configuration as proof of its recognized role within the Federation.

**Trust Mark Issuance to Leaves Affiliated with the Intermediary**

> Trust Marks issued to Leaves (Relying Parties) affiliated with an Intermediary MUST be issued by the Intermediary itself, not directly by the Trust Anchor.

Reviewer: As there are many types of intermediaries I suggest to avoid to use the term "Intermediary" and always refer to the correct one (here and in other places).

Author: Done :).

> Trust Marks issued to Leaves (Relying Parties) affiliated with an Intermediary MUST be issued by the Intermediary itself, not directly by the Trust Anchor. This delegation is made possible by including the Intermediary in the ``trust_mark_issuers`` attribute of the Trust Anchor's Entity Configuration.

Reviewer 2: in un modo o nell'altro è la terza volta che si dice questo

rimuovere le ripetizioni, definire una volta e usare riferimenti

Author: done.

A Relying Party affiliated with an Intermediary MUST include in its own Entity Configuration:

- The Trust Mark issued by the Intermediary, as proof of the successful onboarding process managed by the Intermediary;

> - Its own ``authority_hints`` containing the URL of the Intermediary (not directly of the Trust Anchor), thus indicating the correct hierarchical chain.

Reviewer 2: anche questo ripetuto

in precedenza con MUST, questa volta senza

rimuovere le ripetizioni, ridurre il testo

Author: done.

Below is a non-normative example of a Trust Anchor Entity Configuration showing the updated ``trust_mark_issuers`` configuration with authorized Intermediaries:

.. code-block:: json

    {
      "trust_mark_issuers": {
        "https://trust-anchor.eid-wallet.example.it/trust_marks/federation-entity/openid_credential_verifier": [
          "https://trust-anchor.eid-wallet.example.it",
          "https://intermediary.example.org"
        ],
        "https://trust-anchor.eid-wallet.example.it/trust_marks/federation-entity/rp_intermediary": [
          "https://trust-anchor.eid-wallet.example.it"
        ],
        "https://trust-anchor.eid-wallet.example.it/trust_marks/federation-entity/openid_credential_issuer": [
          "https://trust-anchor.eid-wallet.example.it"
        ],
        "https://trust-anchor.eid-wallet.example.it/trust_marks/federation-entity/wallet_solution": [
          "https://trust-anchor.eid-wallet.example.it"
        ]
      }
    }

.. note::

>   The Trust Mark with identifier ending with ``federation-entity/rp_intermediary`` is issued exclusively by the Trust Anchor and is the one the Wallet uses to verify that an entity is a recognized RP Intermediary. The ``https://<federation_authority_domain>/trust_marks/federation-entity/openid_credential_verifier`` Trust Marks may also be issued by Intermediaries for their affiliated Leaves.

Reviewer 2: ```suggestion
  The Trust Mark with identifier ending with ``federation-entity/rp_intermediary`` is issued exclusively by the Trust Anchor and is the one the Wallet uses to verify that an entity is a recognized RP Intermediary. The ``https://<federation_authority_domain>/trust_marks/federation-entity/openid_credential_verifier`` Trust Marks SHOULD also be issued by Intermediaries for their affiliated Leaves.
```

Reviewer 2: perché era may? Dovrebbe stare in MUST piuttosto

Author: corretto :)
